### PR TITLE
add supply monitoring to backend to support TotalSupply and Rich List  functionality

### PR DESCRIPTION
**Pull request by lazynina**
Note: This is functionality that is required to get DESO verified on CoinMarketCap

### CODE REVIEW

File: routes/supply.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+				fes.RichList = richListResponses
+			case <-fes.quit:
+				break out
+			}
+		}
+	}()
+}
+
+func (fes *APIServer) GetTotalSupply(ww http.ResponseWriter, req *http.Request) {
+	if err := json.NewEncoder(ww).Encode(fes.TotalSupplyDESO); err != nil {
```
Comment (triplegreenshell) [inline routes/supply.go:89]: Should have an error here if supply monitoring is not enabled on this node

File: routes/supply.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		_AddBadRequestError(ww, fmt.Sprintf("GetTotalSupply: Error encoding response: %v", err))
+		return
+	}
+}
+
+func (fes *APIServer) GetRichList(ww http.ResponseWriter, req *http.Request) {
```
Comment (triplegreenshell) [inline routes/supply.go:95]: Error if supply monitoring not enabled

**Review by triplegreenshell — CHANGES_REQUESTED** [review]
Just needs a couple additional api exceptions, new routines look great 💪

**Comment by lazynina** [comment]
converted back to draft while I add in creator coin DESO locked to total supply

**Review by maebeam — APPROVED** [review]

**Pull request merged**